rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java

```diff
@@ -118,40 +119,111 @@ public void testIsLock() {
+        tx2.addInput(txIn);
+        signWithNecessaryKeys(bridgeConstants.getGenesisFederation(), BridgeRegTestConstants.REGTEST_FEDERATION_PRIVATE_KEYS, txIn, tx2, bridgeConstants);
+        assertFalse(BridgeUtils.isPegInTxAndValidateMinimum(tx2, federation, btcContext, bridgeConstants, actForBlock));
         assertFalse(BridgeUtils.isPegInTx(tx2, federation, btcContext, bridgeConstants));
 
+
         // Tx sending 1 btc to the federation, is a lock tx
         BtcTransaction tx3 = new BtcTransaction(params);
         tx3.addOutput(Coin.COIN, federationAddress);
         tx3.addInput(Sha256Hash.ZERO_HASH, 0, new Script(new byte[]{}));
+        assertTrue(BridgeUtils.isPegInTxAndValidateMinimum(tx3, federation, btcContext, bridgeConstants, actForBlock));
         assertTrue(BridgeUtils.isPegInTx(tx3, federation, btcContext, bridgeConstants));
 
+
         // Tx sending 50 btc to the federation, is a lock tx
         BtcTransaction tx4 = new BtcTransaction(params);
         tx4.addOutput(Coin.FIFTY_COINS, federationAddress);
         tx4.addInput(Sha256Hash.ZERO_HASH, 0, new Script(new byte[]{}));
-        assertTrue(BridgeUtils.isPegInTx(tx4, federation, btcContext, bridgeConstants));
+        assertTrue(BridgeUtils.isPegInTxAndValidateMinimum(tx4, federation, btcContext, bridgeConstants, actForBlock));
+
+        // Tx sending btc between old and new value, it is not a lock before iris
+        BtcTransaction tx5 = new BtcTransaction(params);
+        Coin minimumValueBeforeIris = bridgeConstants.getMinimumLockTxValue();
+        Coin valueLock = minimumLockValueAfterIris.plus((minimumValueBeforeIris.subtract(minimumLockValueAfterIris)).div(2));
+        assertTrue(valueLock.isLessThan(minimumValueBeforeIris));
+        tx5.addOutput(valueLock, federationAddress);
+        tx5.addInput(Sha256Hash.ZERO_HASH, 0, new Script(new byte[]{}));
+        assertTrue(BridgeUtils.isValidMinimumPegin(valueLock, bridgeConstants, tx5.getHash(), actForBlock));
+        assertTrue(BridgeUtils.isPegInTx(tx5, federation, btcContext, bridgeConstants));
+        assertTrue(BridgeUtils.isPegInTxAndValidateMinimum(tx5, federation, btcContext, bridgeConstants, actForBlock));
+
+        // Tx sending btc between old and new value, it is a lock after iris
+        when(actForBlock.isActive(ConsensusRule.RSKIP219)).thenReturn(false);
+        BtcTransaction tx6 = new BtcTransaction(params);
+        tx6.addOutput(valueLock, federationAddress);
+        tx6.addInput(Sha256Hash.ZERO_HASH, 0, new Script(new byte[]{}));
+        assertTrue(BridgeUtils.isPegInTx(tx6, federation, btcContext, bridgeConstants));
+        assertFalse(BridgeUtils.isValidMinimumPegin(valueLock, bridgeConstants, tx5.getHash(), actForBlock));
+        assertFalse(BridgeUtils.isPegInTxAndValidateMinimum(tx6, federation, btcContext, bridgeConstants, actForBlock));
     }
 
     @Test
     public void testIsLockForTwoFederations() {
```

Comment:
Same here, these could be several tests in stead of being only one that long

Reply:
In this cases the only modification is to use the new function to resolve the issue. There are not new test cases.